src/jit/codegenxarch.cpp

```diff
@@ -1367,6 +1367,15 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             {
                 assert(!isRegCandidate);
 
+#if defined(FEATURE_SIMD) && defined(_TARGET_X86_)
+                // Loading of TYP_SIMD12 (i.e. Vector3) variable
+                if (treeNode->TypeGet() == TYP_SIMD12)
+                {
+                    genLoadLclTypeSIMD12(treeNode);
+                    break;
+                }
+#endif // defined(FEATURE_SIMD) && defined(_TARGET_X86_)
```

Comment:
On Amd64, there are no TYP_SIMD12 locals since they are normalized to TYP_SIMD16.  TYP_SIMD12 args are actually byrefs and hence accessed using GT_IND.  Therefore, effective spill/reload of TYP_SIMD12 locals doesn't arise.  But spill/reload of TYP_SIMD12 spill temps can still happen.  I see on Amd64, spill/reload spill temps happens by considering them as if they are TYP_SIMD16.  Because these code paths are using emitActualTypeSize() in calls to emitins_R_S/S_R().  As typelist.h, actual type size of TYP_SIMD12 is 16.  This works because while allocating a spill temp of TYP_SIMD12 on amd64, we allocate two stack slots or 16 bytes.  

Look at the following call chains for more details on spill/reload of TYP_SIMD12 spill temps on amd64 in codegenxarch.cpp/codegenlinear.cpp:
genProduceReg() -> rsSpillTree() -> spillReg()
genUnspillRegIfNeeded()

Coming to x86:
we have to handle spill and reload of TYP_SIMD12 locals and also spill temps.  Because existing code paths are under the assumption that GT_LCL_VAR nodes are typed TYP_SIMD8/16/32 and never TYP_SIMD12.
Spill temp spill/reload paths have implicit assumption that TYP_SIMD12 spill temps are allocated 16 bytes on stack.  That would mean tmpPrellocateTemp() logic needs to be modified for x86.


---
In reply to: [87874965](https://github.com/dotnet/coreclr/pull/8109#discussion_r87874965) [](ancestors = 87874965,87874711)